File: CS/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraPivotGrid;

namespace Q205054 {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e) {
            // TODO: This line of code loads data into the 'nwindDataSet.ProductReports' table. You can move, or remove it, as needed.
            this.productReportsTableAdapter.Fill(this.nwindDataSet.ProductReports);

        }

        private void pivotGridControl1_MouseClick(object sender, MouseEventArgs e) {
            PivotGridHitInfo hInfo = pivotGridControl1.CalcHitInfo(e.Location);
            if(hInfo.HitTest == PivotGridHitTest.Value)
                HandleValueMouseClick(hInfo.ValueInfo);
        }

        private void HandleValueMouseClick(PivotFieldValueEventArgs e) {
            PivotGridField[] higherFields = e.GetHigherLevelFields();
            object[] higherValues = new object[higherFields.Length];
            for(int i = 0; i < higherFields.Length; i++)
                higherValues[i] = e.GetHigherLevelFieldValue(higherFields[i]);

            pivotGridControl1.BeginUpdate();
            PivotArea otherArea = GetOtherArea(e);
            List<PivotGridField> otherFields = pivotGridControl1.GetFieldsByArea(otherArea);
            for(int i = 0; i < otherFields.Count; i++) {
                otherFields[i].SortBySummaryInfo.Field = e.FieldCellViewInfo.Item.DataField;
                otherFields[i].SortBySummaryInfo.Conditions.Clear();
                for(int j = 0; j < higherFields.Length; j++) {
                    otherFields[i].SortBySummaryInfo.Conditions.Add(new PivotGridFieldSortCondition(higherFields[j], higherValues[j]));
                }
                if(e.Field != null && e.Field.Area != PivotArea.DataArea)
                    otherFields[i].SortBySummaryInfo.Conditions.Add(new PivotGridFieldSortCondition(e.Field, e.Value));
            }
            pivotGridControl1.EndUpdate();
        }

        private PivotArea GetOtherArea(PivotFieldValueEventArgs e) {
            return e.FieldCellViewInfo.Item.IsColumn ? PivotArea.RowArea : PivotArea.ColumnArea;
        }
    }
}

File: VB/Form1.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraPivotGrid

Namespace Q205054
	Partial Public Class Form1
		Inherits Form

		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles Me.Load
			' TODO: This line of code loads data into the 'nwindDataSet.ProductReports' table. You can move, or remove it, as needed.
			Me.productReportsTableAdapter.Fill(Me.nwindDataSet.ProductReports)

		End Sub

		Private Sub pivotGridControl1_MouseClick(ByVal sender As Object, ByVal e As MouseEventArgs) Handles pivotGridControl1.MouseClick
			Dim hInfo As PivotGridHitInfo = pivotGridControl1.CalcHitInfo(e.Location)
			If hInfo.HitTest = PivotGridHitTest.Value Then
				HandleValueMouseClick(hInfo.ValueInfo)
			End If
		End Sub

		Private Sub HandleValueMouseClick(ByVal e As PivotFieldValueEventArgs)
			Dim higherFields() As PivotGridField = e.GetHigherLevelFields()
			Dim higherValues(higherFields.Length - 1) As Object
			For i As Integer = 0 To higherFields.Length - 1
				higherValues(i) = e.GetHigherLevelFieldValue(higherFields(i))
			Next i

			pivotGridControl1.BeginUpdate()
			Dim otherArea As PivotArea = GetOtherArea(e)
			Dim otherFields As List(Of PivotGridField) = pivotGridControl1.GetFieldsByArea(otherArea)
			For i As Integer = 0 To otherFields.Count - 1
				otherFields(i).SortBySummaryInfo.Field = e.DataField
				otherFields(i).SortBySummaryInfo.Conditions.Clear()
				For j As Integer = 0 To higherFields.Length - 1
					otherFields(i).SortBySummaryInfo.Conditions.Add(New PivotGridFieldSortCondition(higherFields(j), higherValues(j)))
				Next j
				If e.Field IsNot Nothing AndAlso e.Field.Area <> PivotArea.DataArea Then
					otherFields(i).SortBySummaryInfo.Conditions.Add(New PivotGridFieldSortCondition(e.Field, e.Value))
				End If
			Next i
			pivotGridControl1.EndUpdate()
		End Sub

		Private Function GetOtherArea(ByVal e As PivotFieldValueEventArgs) As PivotArea
			Return If(e.IsColumn, PivotArea.RowArea, PivotArea.ColumnArea)
		End Function
	End Class
End Namespace